### DISCUSSION TITLE
Unable to use bindings to get results from previous step

### DISCUSSION THREAD

**Discussion by pradjones** [Budibase/budibase#4650] [Help] [answered]
Hi all,

I am hoping that someone who has got more experience with Budibase will be able to help me with this. I am writing a JS script that needs to use the SQL query results from the previous step. In the past, I was able to run the automation using steps["1"].response to get the result from the previous step. However, this method no longer seems to work for me and I was wondering what the right way to get the results from the previous step would be?

<img width="436" alt="Screen Shot 2022-02-23 at 3 03 42 pm" src="https://user-images.githubusercontent.com/33274485/155274509-987c1298-7a48-4c8f-bfca-58419f38f081.png">
<img width="392" alt="Screen Shot 2022-02-23 at 3 03 12 pm" src="https://user-images.githubusercontent.com/33274485/155274517-7f80467a-0daa-4cac-888f-57e947efb73d.png">

Much appreciated

**Comment by mjashanks** [comment, marked as answer]
Hey @pradjones the syntax is `steps[1].response...`

I realise that the `Bindings` expander suggests that it should be `steps["1"]` - I made this mistake too. Will log an issue for that. Thanks for reporting :)

**Reply by pradjones** [reply]
Hey @mjashanks, thanks for getting back to me. I tried out your suggestion, it seems that it cannot reference this object. 

Even though I have a response object with the items in the info key, I am not able to get the value with steps[1].response.info

<img width="317" alt="Screen Shot 2022-02-23 at 8 26 37 pm" src="https://user-images.githubusercontent.com/33274485/155319389-e7dd4df6-72b5-4bb5-8b53-b8cbd4286728.png">
<img width="307" alt="Screen Shot 2022-02-23 at 8 25 57 pm" src="https://user-images.githubusercontent.com/33274485/155319395-e3b584cb-a186-4d9d-bc73-ddfd9194913d.png">

**Reply by mjashanks** [reply]
Yes - i see this now.

`steps[1].info` is the result of your response. However, for some reason `info` is a JSON object keyed by number (?)

`info` should be an array - each item being a row.

You'll need to convert the JSON object to an array first....maybe do 

```javascript
const response = steps[1].info;
for (let index in response) {
   const thisValue = response[index];
   // use thisValue...
}
```

**Reply by pradjones** [reply]
Thanks for the help @mjashanks! Working perfectly now.
